playground/cdk/CdkSample.AppHost/search.module.bicep

```diff
@@ -0,0 +1,50 @@
+targetScope = 'resourceGroup'
+
+@description('')
+param location string = resourceGroup().location
+
+@description('')
+param principalId string
+
+@description('')
+param principalType string
+
+
+resource searchService_7WkaGluF0 'Microsoft.Search/searchServices@2023-11-01' = {
+  name: toLower(take(concat('search', uniqueString(resourceGroup().id)), 24))
+  location: location
+  tags: {
+    'aspire-resource-name': 'searchService_7WkaGluF0'
+  }
+  sku: {
+    name: 'basic'
+  }
+  properties: {
+    replicaCount: 1
+    partitionCount: 1
+    hostingMode: 'default'
+    disableLocalAuth: true
+  }
+}
+
+resource roleAssignment_7uytIREoa 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  scope: searchService_7WkaGluF0
+  name: guid(searchService_7WkaGluF0.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8ebe5a00-799e-43f5-93ac-243d3dce84a7'))
+  properties: {
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8ebe5a00-799e-43f5-93ac-243d3dce84a7')
+    principalId: principalId
+    principalType: principalType
+  }
+}
+
+resource roleAssignment_QpFzCj55x 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+  scope: searchService_7WkaGluF0
+  name: guid(searchService_7WkaGluF0.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '7ca78c08-252a-4471-8644-bb5ff32d4ba0'))
+  properties: {
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '7ca78c08-252a-4471-8644-bb5ff32d4ba0')
+    principalId: principalId
+    principalType: principalType
+  }
+}
+
+output connectionString string = 'Endpoint=https://${searchService_7WkaGluF0.name}.search.windows.net'
```

Comment:
Can this bet acquired via a property. This may not be valid in non public clouds.

Reply:
Not today - but filed https://github.com/Azure/azure-sdk-for-net/issues/42640 and https://github.com/Azure/azure-sdk-for-net/issues/42639 to do something slightly better than this.  I left a TODO at https://github.com/dotnet/aspire/pull/2820/files#diff-ea51a3d17e41fb2fc8010fc133d0d3ae6ec2f8faa9967ec063ea9208bb027078R71 to track patching this up.